libcontainer/cgroups/fs2/rdma.go

```diff
@@ -0,0 +1,69 @@
+// +build linux
+
+package fs2
+
+import (
+	"strconv"
+	"strings"
+
+	"github.com/opencontainers/runc/libcontainer/cgroups"
+	"github.com/opencontainers/runc/libcontainer/cgroups/fscommon"
+	"github.com/opencontainers/runc/libcontainer/configs"
+)
+
+func isRdmaSet(r *configs.Resources) bool {
+	return len(r.Rdma) > 0
+}
+
+func createCmdString(device string, limits configs.LinuxRdma) string {
+	cmdString := device
+	if limits.HcaHandles != nil {
+		cmdString += " hca_handle=" + strconv.FormatUint(uint64(*limits.HcaHandles), 10)
+	}
+	if limits.HcaObjects != nil {
+		cmdString += " hca_object=" + strconv.FormatUint(uint64(*limits.HcaObjects), 10)
+	}
+	return cmdString
+}
+
+func setRdma(path string, r *configs.Resources) error {
+	if !isRdmaSet(r) {
+		return nil
+	}
+	for device, limits := range r.Rdma {
+		if err := cgroups.WriteFile(path, "rdma.max", createCmdString(device, limits)); err != nil {
+			return err
+		}
+	}
+	return nil
+}
+
+func statRdma(path string, stats *cgroups.Stats) error {
+	if !cgroups.PathExists(path) {
+		return nil
+	}
+	currentData, err := cgroups.ReadFile(path, "rdma.current")
+	if err != nil {
+		return err
+	}
+	currentPerDevices := strings.Split(currentData, "\n")
+	maxData, err := cgroups.ReadFile(path, "rdma.max")
+	if err != nil {
+		return err
+	}
```

Comment:
This string manipulation is probably not very efficient (also in terms of garbage collection).

It's probably easier and better to offload the reading/parsing to `ConvertRDMAEntry` (which I'd rename to `ReadRDMAEntries`).

Something like this
```go
currentEntries, err := fscommon.ReadRDMAEntries(path, "rdma.current")
...
```

and then `ReadRDMAEntries` should look like

```go
func ReadRDMAEntries(dir, file string) ([]cgroups.RdmaEntry, error) {
   fd, err := OpenFile(dir, file)
   if err != nil {
      return nil, err
   }
   defer fd.Close() //nolint:errorlint
   s := bufio.NewScanner(fd)
   for s.Scan() {
   }
   return entries, s.Err()
}
```

This will decrease the amount of temporary objects, unify the processing between v1 and v2 more.

Reply:
@kolyshkin Thanks for this suggestion , added `ReadRDMAEntries` in latest commit.